<!-- This is an example launch file for how a ROS application could use a node -->
<launch>
	<arg name="use_aws_lex" default="false"/>
	<arg name="use_google" default="false"/>
	<arg name="use_chatgpt" default="true"/>
	<arg name="instance_id" default="default"/>
  	<group if="$(arg use_aws_lex)">
		<rosparam file="$(find harmoni_bot)/config/configuration.yaml" subst_value="True"/>
		<param name="instance_id" value="$(arg instance_id)"/>
		<node pkg="harmoni_bot" type="aws_lex_service.py" name="harmoni_bot_lex_$(arg instance_id)" output="screen"  args="$(arg instance_id)"/>
  	</group>
	<group if="$(arg use_google)">
			<rosparam file="$(find harmoni_bot)/config/configuration.yaml" subst_value="True"/>
			<param name="instance_id" value="$(arg instance_id)"/>
			<node pkg="harmoni_bot" type="google_service.py" name="harmoni_bot_google_$(arg instance_id)" output="screen"  args="$(arg instance_id)"/>
  	</group>
	<group if="$(arg use_chatgpt)">
			<rosparam file="$(find harmoni_bot)/config/configuration.yaml" subst_value="True"/>
			<param name="instance_id" value="$(arg instance_id)"/>
			<node pkg="harmoni_bot" type="chatgpt_service.py" name="harmoni_bot_chatgpt_$(arg instance_id)" output="screen"  args="$(arg instance_id)"/>
  	</group>
</launch>
